liesel/model/model.py

```diff
@@ -341,6 +341,37 @@ def build_model(self, copy: bool = False) -> Model:
 
         return model
 
+    def convert_dtype(
+        self, _from: str | jax.numpy.dtype, _to: str | jax.numpy.dtype
+    ) -> GraphBuilder:
+        """
+        Tries to convert the node values in the graph to the specified data type.
+
+        Works for nodes whose value is an array or pytree_. Nodes whose value is of
+        another type are silently ignored.
+
+        .. _pytree: https://jax.readthedocs.io/en/latest/pytrees.html
+        """
+        nodes, _ = self._all_nodes_and_vars()
+
+        def try_astype(x):
+            try:
+                if x.dtype == _from:
+                    return x.astype(_to)
+                else:
+                    return x
+            except AttributeError:
+                return x
+
+        for node in nodes:
+            try:
+                value = jax.tree_map(try_astype, node.value)
+                node.value = value  # type: ignore # data node
+            except AttributeError:
+                pass
```

Comment:
In which cases do we run into an AttributeError here? Only happens, when the node does not have a `.value` attribute, right? When does that happen?

Reply:
I happens if the property doesn't have a setter.

Reply:
I see. Thanks 👍